docs/ai-hint-system.md

```
@@ -0,0 +1,144 @@
+# AI Hint System – Contributor Notes
+
+## Overview
```

markdownlint-cli2 (0.22.1)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)